lib/web/purchases_flutter_web.dart

```diff
@@ -0,0 +1,440 @@
+        case 'checkTrialOrIntroductoryPriceEligibility':
+          return _checkTrialOrIntroductoryPriceEligibility(call.arguments);
+        case 'setProxyURL':
+          return _setProxyURL(call.arguments);
+        case 'canMakePayments':
+          return Future.value(true);
+        case 'isWebPurchaseRedemptionURL':
+          return Future.value(false);
+
+        case 'syncPurchases':
+        case 'collectDeviceIdentifiers':
+        case 'invalidateCustomerInfoCache':
+        case 'setAttributes':
+        case 'showInAppMessages':
+        case 'enableAdServicesAttributionTokenCollection':
+        case 'setSimulatesAskToBuyInSandbox':
+        case 'setAllowSharingStoreAccount':
+          // No-op on web
+          return Future.value();
+
+        case 'purchase':
+        case 'purchaseProduct':
+        case 'recordPurchase':
+        case 'getPromotionalOffer':
+        case 'purchaseDiscountedPackage':
+        case 'purchaseDiscountedProduct':
+        case 'purchaseSubscriptionOption':
+        case 'presentCodeRedemptionSheet':
+        case 'beginRefundRequestForEntitlement':
+        case 'beginRefundRequestForProduct':
+        case 'beginRefundRequestForActiveEntitlement':
+        case 'purchasePackageWithWinBackOffer':
+        case 'purchaseProductWithWinBackOffer':
+        case 'getEligibleWinBackOffersForProduct':
+        case 'redeemWebPurchase':
+          throw UnsupportedPlatformException();
+
+        default:
+          throw UnsupportedError(
+            '${call.method} not implemented on web platform.',
+          );
+      }
+    } catch (e) {
+      rethrow;
+    }
+  }
+
+  Future<void> _setupPurchases(dynamic arguments) async {
+    try {
+      final apiKey = arguments['apiKey'] as String?;
+      if (apiKey == null) {
+        throw PlatformException(
+          code: _configurationErrorCode,
+          message: 'API key is required',
+        );
+      }
+
+      final appUserId = arguments['appUserId'] as String?;
+
+      final options = {
+        'apiKey': apiKey,
+        'appUserId': appUserId,
+        'flavor': _platformName,
+        'flavorVersion': _pluginVersion,
+      };
+
+      _callStaticMethod('configure', [options]);
+    } catch (e) {
+      throw PlatformException(
+        code: _configurationErrorCode,
+        message: 'Purchases SDK not configured. Call configure first.',
+      );
+    }
+  }
+
+  Future<void> _setLogLevel(dynamic arguments) async {
+    final logLevel = arguments['level'] as String;
+    final jsLogLevel = _convertLogLevel(logLevel);
+    _callStaticMethod('setLogLevel', [jsLogLevel]);
+  }
+
+  String _convertLogLevel(String logLevel) {
+    switch (logLevel.toLowerCase()) {
+      case 'verbose':
+        return 'VERBOSE';
+      case 'debug':
+        return 'DEBUG';
+      case 'info':
+        return 'INFO';
+      case 'warn':
+        return 'WARN';
+      case 'error':
+        return 'ERROR';
+      default:
+        return 'SILENT';
+    }
```

Comment:
Once you bump SDK to >= 3.0 you get support for switch expressions, e.g.:
```suggestion
    return switch (logLevel.toLowerCase()) {
      'verbose' => 'VERBOSE',
      'debug' => 'DEBUG',
      'info' => 'INFO',
      'warn' => 'WARN',
      'error' => 'ERROR',
      _ => 'SILENT',
    };
```